metaflow/plugins/kubernetes/kubernetes_job.py

```diff
@@ -53,12 +56,432 @@ def wrapper(*args, **kwargs):
+            jobset_name,
+            main_job_name,
+            main_job_index,
+            main_pod_index,
+            self._kwargs["namespace"],
+        )
+
+        control_job = self.get_replicated_job(
+            "control",
+            jobset_main_addr,
+            subdomain,
+            ubf_context=UBF_CONTROL,
+            node_index=0,
+        )
+
+        worker_jobs = []
+        for i in range(1, num_parallel):
+            worker_job = self.get_replicated_job(
+                "worker{}".format(str(i)),
+                jobset_main_addr,
+                subdomain,
+                task_id_override=self._kwargs["task_id"].replace("control", "worker")
+                + "-"
+                + str(i),
+                ubf_context=UBF_TASK,
+                node_index=i,
+            )
+
+            worker_jobs.append(worker_job)
+
+        self._jobset = jobset.models.jobset_v1alpha2_job_set.JobsetV1alpha2JobSet(
+            api_version=self._group + "/" + self._version,
+            kind="JobSet",
+            metadata=V1ObjectMeta(name=self.name),
+            spec=jobset.models.jobset_v1alpha2_job_set_spec.JobsetV1alpha2JobSetSpec(
+                replicated_jobs=[control_job] + worker_jobs,
+                suspend=False,
+            ),
+        )
+
+    def get_replicated_job(
+        self,
+        job_name,
+        jobset_main_addr,
+        subdomain,
+        task_id_override=None,
+        ubf_context=None,
+        node_index=None,
+    ):
+        client = self._client.get()
+        current_domain = _make_domain_name(
+            self.name,
+            job_name,
+            0,
+            0,
+            self._kwargs["namespace"],
+        )
+        master_port = (
+            int(self._kwargs["port"])
+            if "port" in self._kwargs and self._kwargs["port"]
+            else None
+        )
+
+        job_spec = copy.deepcopy(self._kwargs["job_spec"])
+        job_spec.parallelism = 1
+        job_spec.completions = 1
+
+        job_spec.template.spec.set_hostname_as_fqdn = True
+        job_spec.template.spec.subdomain = subdomain
+        annotations = self._kwargs.get("annotations", {})
+        labels = self._kwargs.get("labels", {})
+        if task_id_override:
+            labels["metaflow/task_id"] = task_id_override
+            job_spec.template.metadata.labels["metaflow/task_id"] = task_id_override
+
+            for idx in range(len(job_spec.template.spec.containers[0].command)):
+                if (
+                    self._kwargs["task_id"]
+                    in job_spec.template.spec.containers[0].command[idx]
+                ):
+                    job_spec.template.spec.containers[0].command[idx] = (
+                        job_spec.template.spec.containers[0]
+                        .command[idx]
+                        .replace(self._kwargs["task_id"], task_id_override)
+                    )
+                # CHECK FOR THE ubf_context in the command.
```

Comment:
Doesn't this have to be done for all replicated jobs, not just the worker jobs (which have task_id_overrides)?